src/test/java/com/johncnstn/auth/security/TokensProviderTest.java

```diff
@@ -0,0 +1,97 @@
+package com.johncnstn.auth.security;
+
+import com.johncnstn.auth.config.AppProperties;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+
+import java.util.Date;
+import java.util.UUID;
+
+import static com.johncnstn.auth.security.DomainGrantedAuthority.USER;
+import static io.jsonwebtoken.SignatureAlgorithm.HS512;
+import static io.jsonwebtoken.io.Decoders.BASE64;
+import static java.lang.System.currentTimeMillis;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+public class TokensProviderTest {
```

Comment:
Why is it not extending one of abstract classes for tests?

Reply:
Why should it be? Do all our unit test really need this?
`AbstractUnitTest` class has quite a lot of overhead including
```
@AutoConfigureMockMvc
@ExtendWith(SpringExtension.class)
-----
@MockBean
protected HttpClient httpClient;
@MockBean
protected HttpResponse<Object> httpResponse;
```

Reply:
1) This parent class is only for pre-config.
@AutoConfigureMockMvc
@ExtendWith(SpringExtension.class) - for example.
Also, mocks will be helpful not in all child classes, but in many.
2) It's much easier to orient in the code.
All tests have to be extended from one of two classes.

Reply:
Done